server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldAndFormat.java

```diff
@@ -9,25 +9,35 @@
 package org.elasticsearch.search.fetch.subphase;
 
 import org.elasticsearch.Version;
-import org.elasticsearch.core.Nullable;
-import org.elasticsearch.common.xcontent.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.logging.DeprecationLogger;
 import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ParseField;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.core.CheckedFunction;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.RestApiVersion;
 
 import java.io.IOException;
 import java.util.Objects;
 
+import static org.elasticsearch.core.RestApiVersion.equalTo;
+import static org.elasticsearch.core.RestApiVersion.onOrAfter;
+
 /**
  * Wrapper around a field name and the format that should be used to
  * display values of this field.
  */
 public final class FieldAndFormat implements Writeable, ToXContentObject {
+    private static final String USE_DEFAULT_FORMAT = "use_field_mapping";
```

Comment:
This class is also used by the `fields` option, in addition to `docvalue_fields`, which means we're allowing `use_field_mapping` to be set on 7.x on the `fields` option, even though it was never allowed before. This is unfortunate, but seems like an okay compromise to me.

Reply:
good point, I was not aware of this.. I don't think the parser has the information about the parent of the current key/value..
The only alternative would be to use two different parsers, but I am not sure if this is not an overdo.
Especially because this is meant to be for rest backwards compatibility. It would be surprising to see someone start using "use_field_mapping" with rest api compatibility for the `fields` when it was not allowed in 7.x..
Also it would not do anything anyway.

Reply:
I also feel like this is the right trade-off to keep the code simple.